### ISSUE TITLE
Type annotations need to be removed from Pulse module.

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#3056]
### What is the expected enhancement?
In order to have documentation that is uniform over all Qiskit elements, the type annotations used in the Pulse module, and related functions, need to be converted into the standard docstring type used in most other modules.

**Comment by lcapelluto** [comment]
@SooluThomas is looking into this on the docs side

**Comment by taalexander** [comment]
See this issue for further discussion https://github.com/Qiskit/qiskit-ibmq-provider/issues/325.

**Comment by ajavadia** [comment]
Yeah would be great if the doc generation could seemlessly handle modules with and without type hints.. @SooluThomas please let us know if you can make this work. @diego-plan9 can you help.

And I agree we need to go towards a unified approach across the whole codebase but I think that decision should not be driven by documentation. It will come down to succinctness of the code vs ease of development, refactoring, and static reasoning. We can have that debate but documentation should really be orthogonal to this.
 
One thing to note would be if the docstrings for the type-hinted modules need to have types repeated again. For example:
```
class A(param1 : Type1):
    """
    Args:
         param1 (Type1)
    """
```

Currently `Type1` is [not repeated in the pulse module](https://github.com/Qiskit/qiskit-terra/blob/529fdbd695f07f73a9d8e28b50958b0b1f2f4040/qiskit/pulse/channels/system_topology.py#L39) docstrings, but [it is in the provider's](https://github.com/Qiskit/qiskit-ibmq-provider/blob/d95979e7dbb94d50c9e9e3bce8aab0c0b48d0e2e/qiskit/providers/ibmq/ibmqprovider.py#L51) docstrings. This has to be consistent too (I know @nonhermitian pointed out google style). I'm fine if we make this decision based on how the docs look.

**Comment by taalexander** [comment]
> Currently Type1 is not repeated in the pulse module docstrings, but it is in the provider's docstrings. This has to be consistent too (I know @nonhermitian pointed out google style). I'm fine if we make this decision based on how the docs look.

[Google-style is that they do not need to be duplicated](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)

**Comment by ajavadia** [comment]
Yes but currently this makes pulse documentation different from the rest of the code. I want to see if autodoc can create uniform documentation regardless of whether the types are in the signature or the docstring.

**Comment by taalexander** [comment]
Agreed. Based on the link above it should be possible, but maybe we are doing something different with our documentation?

**Comment by diego-plan9** [comment]
> Google-style is that they do not need to be duplicated

For the provider, the inclusion of type hints was meant to be tackled incrementally - and is still in progress! The explicit types in the docstrings were consciously left in place, in order to be conservative and decide later in the final stages - I'd personally lean towards pure Google-style too if we manage to tame the doc generation (which hopefully should be possible).

**Comment by SooluThomas** [comment]
I can try to make the doc generation handle modules with and without type hints. Still investigating
the tools that we can use.

**Comment by taalexander** [comment]
Oh, I agree with you @diego-plan9. We should have done this.
